include/flatbuffers/flexbuffers.h

```diff
@@ -1629,6 +1633,223 @@ class Builder FLATBUFFERS_FINAL_CLASS {
   StringOffsetMap string_pool;
 };
 
+// Helper class to verify the integrity of a FlexBuffer
+class Verifier FLATBUFFERS_FINAL_CLASS {
+ public:
+  Verifier(const uint8_t *buf, size_t buf_len, size_t _max_depth = 64,
+           size_t _max_objects = 1000000,
+           // Turning this flag on likely results in faster verification
+           // of larger buffers with many shared keys/strings, but
+           // comes at the cost of a memory allocation 1/8th the size of
+           // the buffer being verified, so is off by default.
+           bool verify_shared_only_once = false)
+      : buf_(buf),
+        size_(buf_len),
+        depth_(0),
+        max_depth_(_max_depth),
+        num_objects_(0),
+        max_objects_(_max_objects) {
+    FLATBUFFERS_ASSERT(size_ < FLATBUFFERS_MAX_BUFFER_SIZE);
+    if (verify_shared_only_once) already_verified_.resize(size_);
```

Comment:
And how is that a problem if the caller was already able to receive an 8x larger buffer in a single buffer? The caller is apparently fine with buffers of that size, so an 1/8th size additional buffer would not be an attack vector on most systems with dynamic memory allocation. I've made it such that this vector can be omitted on memory constrained systems.

Reply:
I am saying even if that someone could make `buf_len = 2^32-1` even if the buffer is not that size, but that would cause the `already_verified_.resize(size_)` to balloon.

Reply:
`buf_len` is supplied by the programmer, not under control of an attacker?

Reply:
Discussed offline.